components/21-atoms/table/table-mobile.functions.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+/* global define, module */
+(function (root, factory) {
+  if (typeof define === 'function' && define.amd) {
+    define(factory);
+  }
+  else {
+    if (typeof exports === 'object') {
+      module.exports = factory();
+    }
+    else {
+      root.MobileTable = factory();
+    }
+  }
+}(this || window, function () {
+  return function (element, options) {
+
+    /**
+     * The table caption.
+     */
+    let caption = null;
+
+    /**
+     * The responsive list version of the table.
+     */
+    let tableList = null;
+
+    /**
+     * Override default options with options param.
+     */
+    options = (() => {
+      const defaults = {
+        headingType: 'h3'
+      };
+
+      const keys = Object.keys(defaults);
+      let newOptions = options || {};
+
+      for (let i = keys.length; i--;) {
+        newOptions[keys[i]] = options[keys[i]] || defaults[keys[i]];
+      }
+
+      return options;
+    })();
+
+    /**
+     * Add a wrapper div with class .table-list as next sibling to the table.
+     */
+    const addTableList = () => {
+      tableList = document.createElement('div');
+      tableList.classList.add('table-list');
+      tableList.setAttribute('aria-labelledby', 'list-description');
+      element.parentNode.insertBefore(tableList, null);
+
+      addList();
+    };
+
+    /**
+     * Adds the individual items to the list <ul></ul> of the table-list wrapper.
+     *
+     * @param {Object} list
+     *   The list to add in elements based on the table.
+     */
+    const addListItems = (list) => {
+      // Get the headers of the table columns.
+      let colHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="col"');
+      // Get the headers of the table rows.
+      let rowHeadingsNodeList = element.querySelector('table').querySelectorAll('th[scope="row"]');
+      // Get all rows of the table.
+      let rowsNodeList = element.querySelector('table').querySelectorAll('tr');
+      // The first row of the table.
+      const firstRow = element.querySelector('tr');
+
+      // Determine if we have column headings.
+      options.columnHeaders = (firstRow.getElementsByTagName('th').length === firstRow.querySelectorAll('*').length);
+
+      // Determine if we have column headers. If so set the initial row +1.
+      let initial = (options.columnHeaders) ? 1 : 0;
```

Comment:
Unnecessary parentheses + 

```suggestion
      let initial = options.columnHeaders ? 1 : 0;
```

**Sidenote:** Might be worth investigating whether the [Number](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number) object can be used. This isn't natively supported by IE11, but not sure if Babel transpiles this.... 🤔

```suggestion
      let initial = Number(options.columnHeaders);
```